### ISSUE TITLE
Enable node shutdown API for masters when gateway.recover_after_data_nodes is set

### ISSUE THREAD

**Issue by nemonster** [elastic/elasticsearch#10734]
After setting the gateway.recover_after_data_nodes any started masters will not respond to _shutdown.  It will result in:
```
{ 
   "error" : "ClusterBlockException[blocked by: [SERVICE_UNAVAILABLE/1/state not recovered /    initialized];]", 
   "status" : 503 
}
```

In this case the decision has been made to take down the cluster before any data nodes are started so is it necessary to continue to block operations in this case?

**Comment by clintongormley** [comment]
Interestingly, there is talk of removing the shutdown API altogether...  @s1monw ?

**Comment by s1monw** [comment]
YEAH IMO we should remove the shutdown API completely without replacement. This is what OS scripts are for

**Comment by s1monw** [comment]
see https://github.com/elastic/elasticsearch/pull/10831

**Comment by pickypg** [comment]
This is closed by #10831.
